pkg/telemetryservice/minio/minio.go

```diff
@@ -0,0 +1,119 @@
+package minio
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/edgenesis/shifu/pkg/deviceshifu/deviceshifubase"
+	"github.com/edgenesis/shifu/pkg/deviceshifu/deviceshifuhttp"
+	"github.com/edgenesis/shifu/pkg/k8s/api/v1alpha1"
+	"github.com/edgenesis/shifu/pkg/logger"
+	"github.com/edgenesis/shifu/pkg/telemetryservice/utils"
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"io"
+	"net/http"
+	"time"
+)
+
+func BindMinIOServiceHandler(w http.ResponseWriter, r *http.Request) {
+	// Parse request content
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		logger.Errorf("Error when Read Data From Body, error: %v", err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	request := v1alpha1.TelemetryRequest{}
+
+	err = json.Unmarshal(body, &request)
+	if err != nil {
+		logger.Errorf("Error to Unmarshal request body to struct")
+		http.Error(w, "Unexpected end of JSON input", http.StatusBadRequest)
+		return
+	}
+	if request.MinIOSetting.Bucket == nil || request.MinIOSetting.EndPoint == nil || request.MinIOSetting.FileExtension == nil {
```

Comment:
we can also add `+kubebuilder:validation` in `telemetryservice_types.go` to let CRD validate these